src/compiler/parser.ts

```diff
@@ -4372,6 +4372,12 @@ module ts {
 
         function parsePropertyOrMethodDeclaration(fullStart: number, modifiers: ModifiersArray): ClassElement {
             var asteriskToken = parseOptionalToken(SyntaxKind.AsteriskToken);
+
+            // From ES6 Specification, "implements", "interface", "let", "package", "private", "protected", "public", "static", and "yield" are reserved words within strict mode code

```

Comment:
I don't think this is correct.  Yes, they are reserved words, however a property/method is allowed to use reserved words as its name.  

```
PropertyName[Yield,GeneratorParameter] :
  LiteralPropertyName
  [+GeneratorParameter] ComputedPropertyName
  [~GeneratorParameter] ComputedPropertyName[?Yield]
See 12.2.5
LiteralPropertyName :
  IdentifierName
  StringLiteral
  NumericLiteral
```

All keywords are valid IdentifierNames, so we should continue allowing keywords here, even in strict mode.